gamestonk_terminal/portfolio_optimization/port_optimization_controller.py

```diff
@@ -0,0 +1,277 @@
+        self.tickers = set(tickers)
+        # These will allow the ca menu to be re-access
+        self.ca_ticker = None
+        self.ca_similar = None
+
+    @staticmethod
+    def print_help(tickers: List[str]):
+        """Print help"""
+        print("\nPortfolio Optimization:")
+        print("   help          show this menu again")
+        print("   q             quit this menu, and shows back to main menu")
+        print("   quit          quit to abandon program")
+        print("   > ca          comparison analysis menu")
+        print("   > scr         screener menu")
+        print(f"\nCurrent Tickers: {('None', ', '.join(tickers))[bool(tickers)]}")
+        print("")
+        print("   add          add ticker to optimize")
+        print("   select       overwrite current tickers with new tickers")
+        print("")
+        print("Optimization:")
+        print("")
+        print("   Property weighted:")
+        print("       equal_weight   equally weighted portfolio")
+        print("       mkt_cap        marketcap weighted portfolio")
+        print("       div_yield      dividend weighted portfolio\n")
+        print("   Mean Variance Optimization :")
+        print("        max_sharpe    portfolio with maximum sharpe ratio")
+        print("        min_vol       portfolio with minimum volatility")
+        print("        eff_risk      portfolio that maximizes returns at given risk")
+        print("        eff_ret       portfolio that minimizes risk at given return")
+        print("        show_ef       show the efficient frontier")
+        print("")
+        plt.close("all")
+
+    def switch(self, an_input: str):
+        """Process and dispatch input
+
+        Returns
+        -------
+        True, False or None
+            False - quit the menu
+            True - quit the program
+            None - continue in the menu
+        """
+        (known_args, other_args) = self.po_parser.parse_known_args(an_input.split())
+
+        return getattr(
+            self, "call_" + known_args.cmd, lambda: "Command not recognized!"
+        )(other_args)
+
+    def call_help(self, _):
+        """Process Help command"""
+        self.print_help(self.tickers)
+
+    def call_q(self, _):
+        """Process Q command - quit the menu"""
+        return False
+
+    def call_quit(self, _):
+        """Process Quit command - quit the program"""
+        return True
+
+    def call_ca(self, _):
+
+        return ca_controller.menu(pd.DataFrame(), self.ca_ticker, "", "1440min")
+
+    def call_scr(self, _):
+        return screener_controller.menu()
+
+    def call_add(self, other_args: List[str]):
+        self.add_stocks(self, other_args)
+
+    def call_select(self, other_args: List[str]):
+        self.tickers = set([])
+        self.add_stocks(self, other_args)
+
+    def call_equal_weight(self, other_args: List[str]):
+        weights = po_api.equal_weight(self.tickers, other_args)
```

Comment:
To keep the structure the same, could you have this using argparse?

This way the user can add the -h flag and we can add a nice description to what the optimization does. Also, in the future we may need to add arguments for these, so this way it would be more future proof.

Another thing is that it would be nice to see bars with percentage, and even have an option that allows for discrete allocation based on a pre-defined user allocation.

Reply:
So have an argument that would either be like --marketcap or --dividend instead of individual functions (where i input the property type arg)?

Reply:
No. Having it as you have, but parsing it through an argparse, so that the user can do:
```
market_cap -h
```
and get a description of what the command does. 

And in the future if we add that "value component" that the user can add, it's gonna be easier as the user can do:
```
market_cap --value 1000
```
and instead of getting: FB 0.5, AMZN 0.5, he gets: FB 500, AMZN 500. If that makes sense.

Also, the current outputs are not the prettiest, as you can still see the brackets. I would put these on a dataframe, and output:
```
ticker    value
AMZN    50%
FB          50%
```

Reply:
oh yeah that should be straightforward.  I was thinking there could be something like --value or --shares that would return the amount.  I also just printed the dictionary like in the pyportfolioopt module.

In terms of plotting, I was thinking I could default to a pie chart or something (in the case of long only) but have options for others